### DISCUSSION TITLE
x-model programmatic api broken

### DISCUSSION THREAD

**Discussion by kastriotkastrati** [alpinejs/alpine#2683] [7. Bugs (archived)] [answered]
Hello there,

using any x-model inside an `x-modelable` or the programmatic api affects the `modelable` variable.

```
<div x-data="{isOpen:false}">
<div x-data="{ 
             open: false,
             text: ''
             }" x-modelable='open' x-model='isOpen'>
    <button @click="open = true">Expand</button>
 
    <div x-show="open">
      open: <span x-text='open'></span>
      <br/>
      <input type='text' value='default' x-model='text'/>
      
    </div>
  </div>
</div>
```

demo:
https://jsfiddle.net/hp6qv1mb/2/

**Comment by SimoTod** [comment, marked as answer]
It's counterintuitive but expected.
The inner x-model will trigger an input event, that event will bubble up and eventually hit the element with the outer x-model triggering the external update. 
x-modelable is not that relevant, it just makes it more obvious, the event will change the outer x-model regardless.

For example, you can put `@input.stop` on the inner x-model to prevent the event from bubbling up.

P.s. you created a duplicate in the issues section, this project doesn't use it: see https://twitter.com/calebporzio/status/1430564014464159749
